pkg/api/trillian_client.go

```diff
@@ -76,6 +77,52 @@ func (t *TrillianClient) addLeaf(byteValue []byte) *Response {
 	}
 	resp, err := t.client.QueueLeaf(t.context, rqst)
 
+	// check for error
+	if err != nil || (resp.QueuedLeaf.Status != nil && resp.QueuedLeaf.Status.Code != int32(codes.OK)) {
+		return &Response{
+			status:       status.Code(err),
+			err:          err,
+			getAddResult: resp,
+		}
+	}
+
+	root, err := t.root()
+	if err != nil {
+		return &Response{
+			status:       status.Code(err),
+			err:          err,
+			getAddResult: resp,
+		}
+	}
+	verifier, err := client.NewLogVerifierFromTree(t.tree)
```

Comment:
nit: is there a way we could create this once and cache it on the t object? It doesn't appear to rely on anything in the request context.

Reply:
yup, will fix